### DISCUSSION TITLE
apply query filter on object type

### DISCUSSION THREAD

**Discussion by rehmaninvozone** [prisma/prisma#16764] [Q&A] [answered]
Hi, I need help. I want to know how can I apply query filter on object type field.
I have a user model something like this 
```
model User {
  id String  @id @default(auto()) @map("_id") @db.ObjectId
  firstName String @default("Anonymous")
  lastName String?
  email String @unique
  password  String?
  avatar String?
  companies UserCompany? 
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  @@map("users")
}
```
and also have a user company type

```
type UserCompany {
  companyId String?
  systemRoleId String? 
}
```
To apply the query filter on companies it should be unique like   ```companies UserCompany? @unique``` but this is not working I am getting an error. any help? thanks

**Comment by nurul3101** [comment, marked as answer]
Hello @rehmaninvozone 👋 
Can you please elaborate on what you are trying to achieve?
`companies UserCompany? @unique` what does this statement intend to do?
UserCompany is a composite type and not a scalar type, do you want to enforce a `@unique` constraint on fields of UserCompany type? 
And what error are you getting?

**Reply by rehmaninvozone** [reply]
Hi, thanks for your reply. What I want is to do something like this

```
this.prisma.user.update({
      data: newUserData,
      where: {
        id: userId,
        companies: { is: { companyId } } 
      //  or if possible I'm new to prisma don't know the exact syntax
        companies.companyId: companyId
      },
    });
```
    
yes, UserCompany is a composite type so @unique not working on it.

**Reply by rehmaninvozone** [reply]
Hi, @nurul3101 can you please help with this?

**Reply by nurul3101** [reply]
Here's a working example which should unblock you

I am creating a record and updating it based on where condition of composite type.
You would need to enable [extendedWhereUnique](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#filter-on-non-unique-fields-with-userwhereuniqueinput) preview feature for this.

schema.prisma
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["extendedWhereUnique"]
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id        String       @id @default(auto()) @map("_id") @db.ObjectId
  firstName String       @default("Anonymous")
  lastName  String?
  email     String       @unique
  password  String?
  avatar    String?
  companies UserCompany?
  createdAt DateTime     @default(now())
  updatedAt DateTime     @updatedAt

  @@map("users")
}

type UserCompany {
  companyId    String?
  systemRoleId String?
}

```

index.ts
```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({});

async function main() {
  const MongoId = '5f9b9b9b9b9b9b9b9b9b9b9b';
  await prisma.user.create({
    data: {
      id: MongoId,
      email: 'test@prisma.io',
      avatar: 'https://prisma.io',
      companies: {
        companyId: '1',
        systemRoleId: '1',
      },
      firstName: 'Test',
      lastName: 'User',
    },
  });

  const updatedResponse = await prisma.user.update({
    where: {
      id: MongoId,
      companies: {
        companyId: '1',
        systemRoleId: '1',
      },
    },
    data: {
      firstName: 'Updated NAME 2',
    },
  });

  console.log(updatedResponse);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

Here's the response:
<img width="613" alt="Screenshot 2022-12-14 at 6 15 54 PM" src="https://user-images.githubusercontent.com/13049676/207598810-5707f5f7-a1e0-40a4-aa2d-43900868bf9b.png">

**Reply by rehmaninvozone** [reply]
oh so, I just need to enable ```extendedWhereUnique``` preview feature for this. Thank you for your help.

**Reply by nurul3101** [reply]
Yes, you would need to enable extendedWhereUnique preview feature. No worries 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
